### PR TITLE
Add marginBottom to Table to break before reaching the bottom of the page, in order to have a footer

### PR DESCRIPTION
**Pull request by jjchiw**
Hi

I added a property to `table` `marginBottom` and what it does is that it would _break_ and go the next page, I needed that value because I have a _footer_ in every page and if the table has many rows the _row_ and _footer_ overlaps.

Thanks for the library

### CODE REVIEW

**Comment by LuckyForce** [comment]
Same Thing for the Header. I just saw your pull request here. I opened another issue for Header.

File: index.js [hunk 1]
```diff
@@ -472,9 +472,9 @@ class PDFDocumentWithTables extends PDFDocument {
           }
 
           // 24.1 is height calc title + subtitle
-          titleHeight = !lockAddTitles ? 24.1 : 0; 
+          titleHeight = !lockAddTitles ? 24.1 : 0;
           // calc if header + first line fit on last page
-          const calc = startY + titleHeight + firstLineHeight + this.headerHeight + safelyMarginBottom// * 1.3;
+          const calc = startY + titleHeight + firstLineHeight + this.headerHeight + options.marginBottom// * 1.3;
```
Comment (LuckyForce) [inline index.js:477]: Are you not already adding the marginBottom to the safelyMarginBottom. Wouldnt you want both to be considerated. ergo not changing this line

File: index.js [hunk 2]
```diff
@@ -784,79 +785,82 @@ class PDFDocumentWithTables extends PDFDocument {
...
             prepareRow(row, index, i, rectRow, rectCell);
-    
+
             if(typeof table.headers[index] === 'object') {
               // renderer column
               table.headers[index].renderer && (cell = table.headers[index].renderer(cell, index, i, row, rectRow, rectCell, this)); // text-cell, index-column, index-line, row, doc[this]
               // align
               table.headers[index].align && (align = table.headers[index].align);
               table.headers[index].valign && (valign = table.headers[index].valign);
+              table.headers[index].fontName && (fontName = table.headers[index].fontName);
```
Comment (LuckyForce) [inline index.js:858]: You would already set the font on prepareRow

**Comment by jjchiw** [comment]
Hi, uff I went to quickly to make the PR, I'm still changing some stuff in my `repo` I'll made another `pr` 

Thanks :)

**Comment by LuckyForce** [comment]
No problem, but you dont necessary need to close the pull request instantly just add new commits to it

**Pull request closed without merging**